lib/OpenQA/WebAPI/Controller/Admin/Workers.pm

```diff
@@ -109,5 +109,25 @@ sub previous_jobs_ajax {
     );
 }
 
+sub delete{
+    my ($self) = @_; 
+    my $w = $self->db->resultset('Workers')->find($self->param('worker_id'));
+
```

Comment:
I'd prefer a proper variable name, e.g. `$worker`. I guess you should also handle the case when `$self->param('worker_id')` returns `undef`.

---

The if-else-cascade is better written as:

```
if (!$worker) {
    $self->flash(error => 'Worker not found.');
    return;
}
if (!($w->status eq 'dead' && !$w->job)) {
    $self->flash(error => 'Worker not found.');
    return;
}
```  
followed by the 'happy path'.

Reply:
This is a case where i would like to see us use proper validation.
```perl
 my $validation = $self->validation;
$validation->required('worker_id')->num;
return $self->render(text => 'Bad request', status => 400) if $validation->has_error;

my $worker_id = $validation->param('worker_id');
```

Reply:
Btw. In #2029 i've also added a helper that makes nicer error messages than `Bad request`. You can use it with `return $self->reply->validation_error if $validation->has_error;`.